DataFormats/Math/interface/Graph.h

```diff
@@ -344,12 +340,10 @@ void graph<N,E>::findRoots(edge_list & result) const
   
   for (; it != ed; ++it) {
     const edge_list & el = *it;
-    typename edge_list::const_iterator el_it = el.begin();
-    typename edge_list::const_iterator el_ed = el.end();
-    for (; el_it != el_ed; ++el_it) {
-      rootCandidate[el_it->first]=false; 
+    for (auto const & el_it : el) {
+      rootCandidate[el_it.first]=false; 
       //el_rt = el_it; // stop at the first encountered candidate!
-      //std::cout << "graphwalker: found a root candidate = " << g.nodeData(el_rt->first) << std::endl;
+      //std::cout << "Graphwalker: found a root candidate = " << g.nodeData(el_rt->first) << std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

this comment applies to a few more places in this file